jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/internal/FrameFlusher.java

```diff
@@ -511,27 +516,61 @@ public String toString()
         return String.format("%s[queueSize=%d,aggregate=%s]",
             super.toString(),
             getQueueSize(),
-            batchBuffer);
+            _batchBuffer);
     }
 
-    private static class Entry extends FrameEntry
+    private class Entry extends FrameEntry implements CyclicTimeouts.Expirable
     {
-        private final long timeOfCreation = System.currentTimeMillis();
+        private long _expiry = Long.MAX_VALUE;
```

Comment:
`_expiry` is only set in t he constructor, you should do everything you can to make it `final`, i.e.: don't give it a default value here, but use a temp variable in the constructor that you override depending on the config and finally set that temp variable to the field.